src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -484,6 +485,69 @@ public boolean impliesClusterPermissionPermission(String action) {
             return roles.stream().filter(r -> r.impliesClusterPermission(action)).count() > 0;
         }
 
+        /**
+         * Checks if the route is accessible via legacy naming convention.
+         * We check against cluster permissions because the legacy convention for cluster permissions map 1-1 with a transport
+         * action call initiated via REST API handler. Hence we use the same to allow/block request forwarding to ext.
+         * This ensures backwards-compatibility
+         *
+         *
+         * NOTE: THIS CHECK WILL BE REMOVED ONCE ALL ACTIONS HAVE BEEN MIGRATED TO THE NEW CONVENTION
+         *
+         * E.g For ext `hw`, following are two possible ways actions an be defined in roles:
+         *
+         * ext_hw_greet:
+         *   reserved: true
+         *   cluster_permissions:
+         *     - 'hw:greet'
+         *
+         * legacy_hw_greet:
+         *   reserved: true
+         *   cluster_permissions:
+         *     - 'cluster:admin/opensearch/hw/greet'
+         *
+         *
+         * @param action - The action to be checked against its legacy version
+         * @return true, if a legacy version was found and validated, false otherwise
+         */
+        @Override
+        public boolean impliesLegacyPermission(String action) {
+            boolean isAlreadyLegacy = action.startsWith("cluster:admin/open");
+            if (isAlreadyLegacy) {
+                return false; // this check was already made, so return false
+            }
+
+            log.info("Checking legacy permissions for {}", action);
+
+            action = action.split(":")[1]; // e.g. `hw:greet` would check for action `greet` for ext `hw`
+
+            /* Regex: `/(?:cluster:admin\/\b(open(distro|search))\b\/[a-zA-Z]+\/|\*)action\/?(?:\*|[\/a-zA-Z0-9]*)/gm`
+             *  matches:
+             *  *action*
+             *  cluster:admin/opensearch/abcd/action
```

Comment:
My main issue with this design is that it implies that even newly minted routes will have a legacy name associated with them. 

Is it possible to have the flexibility to let a plugin developer who is converting their plugin to an extension the power to define an old action name (the action name as it is in the plugin) and a newer name where they are not constrained to the `cluster:*` naming convention of a transport action?

Reply:
Looking at AD, the current transport action name for get detector info is `cluster:admin/opendistro/ad/detector/info`.

The convention put forward in the SDK in this PR (https://github.com/opensearch-project/opensearch-sdk-java/pull/622 - its been sitting for a while) would use the `extensionShortName` to prefix the names of routes. 

For HelloWorld extension that would mean that if the short name of the extension is `hw` then all names of (REST) actions would be prefixed with `hw:`. Granting full access to the extension would be `hw:*` or granting individual actions with `hw:greet` for example. `hw` is defined in `extensions.yml` though and could be changed to `hello`.

In this design it looks like it would be required for AD as an extension to have a shortened name of `ad` and would not be able to be changed to something like `anomaly`. That may be ok, but would be great if it had the flexibility.

Reply:
For now, I have made the checks plugin name agnostic. As long as the unique action name matches, it will return true.